components/rdf-ntriples.js

```diff
@@ -0,0 +1,86 @@
+// rdf-ntriples.js
+
+var _ = require('underscore');
+
+var promiseOutput = require('../src/promise-output');
+var componentFactory = require('../src/noflo-component-factory');
+
+/**
+ * Converts an RDF JS Interface Graph object into an object with the property
+ * 'tokens' containing an Array of N-Triples tokens. tokens.join('') will produce an N-Triples string.
+ */
+exports.getComponent = componentFactory({
+    description: "Converts an RDF JS Interface Graph object into an Array of N-Triples tokens",
+    icon: 'edit',
+    outPorts: promiseOutput.outPorts,
+    inPorts: {
+        input: {
+            description: "RDF JS Interface Graph object",
+            datatype: 'object',
+            required: true,
+            ondata: promiseOutput(execute)
+        }
+    }
+});
+
+/**
+ * Promise of N-Triples array of tokens for the given graph.
+ * @this a noflo.InPort or facade
+ * @param graph RDF JS Interface Graph object
+ */
+function execute(graph) {
+    var tokens = [];
+    graph.forEach(function(triple) {
+        tokens.push.apply(tokens, valueTokens(triple.subject));
+        tokens.push(' ');
+        tokens.push.apply(tokens, valueTokens(triple.predicate));
+        tokens.push(' ');
+        tokens.push.apply(tokens, valueTokens(triple.object));
+        tokens.push(' .', '\n');
+    });
+    return {tokens: tokens};
+}
+
+/**
+ * Converts the RDF term into a array of tokens
+ * @param object an RDF term implementing RDF JS Interface
+ */
+function valueTokens(term) {
+    var value = term.nominalValue;
+    var encoded = encodingNeededFor(value) ? encode(value) : value;
+    if (term.interfaceName === 'NamedNode') {
+        if (encoded != value || value.indexOf('>') >= 0)
+            throw Error("Invalid NamedNode: " + value);
+        return ['<', encoded, '>'];
+    } else if (term.interfaceName === 'BlankNode') {
+        if (encoded != value || value.indexOf(' ') >= 0)
+            throw Error("Invalid BlankNode: " + value);
+        return ['_:', encoded];
+    } else if (term.language) {
+        if (encodingNeededFor(term.language))
+            throw Error("Invalid Language: " + value);
+        return ['"', encoded, '"@', term.language];
+    } else if (term.datatype && term.datatype.toString() != 'http://www.w3.org/2001/XMLSchema#string') {
+        var datatype = term.datatype.toString();
+        if (encodingNeededFor(datatype) || datatype.indexOf('>') >= 0)
+            throw Error("Invalid Datatype: " + datatype);
+        return ['"', encoded, '"^^<', datatype, '>'];
+    } else {
+        return ['"', encoded, '"'];
+    }
+}
+
+/**
+ * Checks if the string includes any special characters.
+ */
+function encodingNeededFor(st) {
+    return st.indexOf('\\') >= 0 || st.indexOf('\"') >= 0 ||
```

Comment:
you spell it out as string rather than st most other places - can we keep with that so it's a no-brainer at first glance as to what that parameter is?